app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -71,14 +71,17 @@ def sso_authentication
 
   def set_current_user(user)
     User.current = user
-
     # API access resets the whole session and marks the session as initialized from API
     # such sessions aren't checked for CSRF
     # UI access resets only session ID
     if api_request?
+      jwt_data = { sso_method: session[:sso_method], expires_at: session[:expires_at] }
```

Comment:
Why is this only for API requests?
I believe this does not belong to `set_current_user`.

Reply:
still not addressed

Reply:
@tbrisker in the previous comments I had mentioned the reason for this and it seemed reasonable, we are doing a `reset_session` on line [79](https://github.com/theforeman/foreman/pull/6549/files#diff-f41d9ad6a64c4df8265f8794634c27beR79) and therefore we would require this line of code here.